### ISSUE TITLE
[JENKINS-38398] Parallel branches do not show completed in the stage graph when the branch is finished

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3268]
<p>This is a follow on from: <a href="https://issues.jenkins-ci.org/browse/JENKINS-37962" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-37962</a></p>

<p>which is currently blocked by: <a href="https://issues.jenkins-ci.org/browse/JENKINS-38223" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-38223</a></p>

<p>(or a change to the back end). There is likely no front end changes needed for this, but this is a place holder to track that external issue. </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-38398" target="_blank">Parallel branches do not show completed in the stage graph when the branch is finished</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2016-10-24T02:57:31+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>This is a follow on from: <a href="https://issues.jenkins-ci.org/browse/JENKINS-37962" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-37962</a></p>

<p>which is currently blocked by: <a href="https://issues.jenkins-ci.org/browse/JENKINS-38223" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-38223</a></p>

<p>(or a change to the back end). There is likely no front end changes needed for this, but this is a place holder to track that external issue. </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-38398?focusedId=271031&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-271031">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Need to wait for <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> to implement <a href="https://issues.jenkins.io/browse/JENKINS-37667" title="Use ForkScanner to build DAG (ie adapt to bismuth api)" class="issue-link" data-issue-key="JENKINS-37667"><del>JENKINS-37667</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Need to wait for <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> to implement <a href="https://issues.jenkins.io/browse/JENKINS-37667" title="Use ForkScanner to build DAG (ie adapt to bismuth api)" class="issue-link" data-issue-key="JENKINS-37667"><del>JENKINS-37667</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-38398?focusedId=274547&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-274547">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Likely the same as  <a href="https://issues.jenkins.io/browse/JENKINS-38536" title="Execution time of parallel blocks of in-flight jobs wrong" class="issue-link" data-issue-key="JENKINS-38536"><del>JENKINS-38536</del></a></p>
</pre>
</details>

</ul>
</details>

<p>Likely the same as  <a href="https://issues.jenkins.io/browse/JENKINS-38536" title="Execution time of parallel blocks of in-flight jobs wrong" class="issue-link" data-issue-key="JENKINS-38536"><del>JENKINS-38536</del></a></p>
